dissect/ffs/ffs.py

```diff
@@ -79,27 +79,35 @@ def cylinder_groups(self):
             yield self.cylinder_group(num)
 
     @lru_cache(4096)
-    def inode(self, inum, name=None, filetype=None):
-        return INode(self, inum, name, filetype)
+    def inode(self, inum, name=None, filetype=None, parent=None):
+        return INode(self, inum, name, filetype, parent=parent)
 
     def get(self, path, node=None):
         if isinstance(path, int):
             return self.inode(path)
 
-        node = node or self.root
+        path = path.replace("\\", "/")
+        node = node if node else self.root
 
         parts = path.split("/")
-        for i, p in enumerate(parts):
-            if not p:
+
+        for part_num, part in enumerate(parts):
+            if not part:
                 continue
 
-            for child in node.iterdir():
-                if child.name == p:
-                    node = child
-                    break
-            else:
+            while node._type == stat.S_IFLNK and part_num < len(parts):
+                node = node.link_inode
+
+            dirlist = node.listdir()
+
+            # there can be residual nul bytes in the file name
+            part = part.replace("\x00", "")
```

Comment:
```suggestion
```
This is a bug here: https://github.com/fox-it/dissect.ffs/pull/11/files#diff-1769ec83c9bd94fdb702b532a8dc95111a2d99a39d301c227e1056a56a8bf265R312-R321

I can't place a comment there because GitHub doesn't let me, but the issue is that "small files" get the entirety of those arrays written to a `BytesIO` object, while it should be restricted to `self.size` amount of bytes.

Can you fix that instead of applying a band-aid here?

Reply:
Applied